packages/react-dom/src/__tests__/ReactDOMForm-test.js

```diff
@@ -1670,6 +1670,48 @@ describe('ReactDOMForm', () => {
     expect(divRef.current.textContent).toEqual('Current username: acdlite');
   });
 
+  it('multiple form submissions in rapid succession do not throw', async () => {
+    const submitFormTwiceButtonRef = React.createRef();
+    let actionCounter = 0;
+    function App() {
+      const formRef = React.createRef();
+      // Submits the form twice in quick succession
+      // This simulates a user clicking a button twice in rapid succession
+      async function submitFormTwice() {
```

Comment:
Instead of doing this in the component, can you do it outside, like doing:

```
await act(async () => {
      submitFormTwiceButtonRef.current.click();
      await waitFor(['Render']);
      submitFormTwiceButtonRef.current.click();
});
```
And a `Scheduler.log('Render');` in render.

This will test closer to what a real user interaction would be like.

Reply:
I changed await `waitFor(['Render']);` to `waitForMicrotasks()`. Thoughts?

Reply:
that's fine too